geo/src/algorithm/euclidean_distance.rs

```diff
@@ -585,8 +586,18 @@ pub fn nearest_neighbour_distance<T>(geom1: &LineString<T>, geom2: &LineString<T
 where
     T: GeoFloat + RTreeNum,
 {
-    let tree_a: RTree<Line<_>> = RTree::bulk_load(geom1.lines().collect::<Vec<_>>());
-    let tree_b: RTree<Line<_>> = RTree::bulk_load(geom2.lines().collect::<Vec<_>>());
+    let tree_a = RTree::bulk_load(
+        geom1
+            .lines()
+            .map(CachedEnvelope::new)
+            .collect::<Vec<_>>(),
```

Comment:
```suggestion
            .collect(),
```

Sorry :sweat_smile:.